### PR TITLE
revamp public map [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by matallo**
first commit, familiarizing myself with the code and so, just made the changes in the header for now

- created a `dashboard_header`, do we want to change them only in the public pages or in the dashboard , too? Inevitably the styles will change

screen shot 2015-07-07 at 16 48 29

then, already in the `public_header` 

- left the links with a bit of opacity to keep the hover states, didn't found them in the mockups, **is it white with underline, do I leave them like this**?

- do we lose "Learn" in favour of "Industries" + "Pricing" ?
- I suppose the link "Login" was left with no update, applied the same style **please confirm you want it different**

screen shot 2015-07-07 at 16 46 47

- had to modify `.UserAvatar-img` in order to the signup button to be aligned. Avatar is 42px with that transparent border (what is it for?), button is 44px :/

- despite we already have a size variable for `40px` I created a new one for the items separation, modifying `$sMargin-element` wasn't feasible, so many side changes.

- finally, I created `.Header-navigationItem` as we have `.Header-settingsItem`, actually, don't you think we should consolidate them both in a single class? the same with `.Header-navigationLink`

will go on with this, but as I first dove here I prefer to check these before going any further ^^^

/cc @CartoDB/frontend

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4768/))

**Comment by matallo** [comment]
ping @viddo @javierarce (as you are the most concerned with styleguide and @xavijam specifically asked me to ping other people in this case)

File: app/assets/stylesheets/variables/sizes.css.scss [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
-$sMargin-section:       40px; // To separate main section
-$sMargin-group:         28px; // To separate groups inside a section
-$sMargin-element:       20px; // To separate elements inside a group
-$sMargin-elementInline: 12px; // To separate inline elements
-$sMargin-min:           4px;  // Minimum margin, special cases only
-$sMargin-formRow:       20px; // Each row in a form should have a margin-bottom of this value.
+$sMargin-section:            40px; // To separate main section
+$sMargin-group:              28px; // To separate groups inside a section
+$sMargin-element:            20px; // To separate elements inside a group
+$sMargin-elementLarge:       40px; // To separate elements inside a group (large separation)
```
Comment (viddo) [inline app/assets/stylesheets/variables/sizes.css.scss]: Why not use `$sMargin-section`?

**Comment by viddo** [comment]
I answer the questions I feel I can answer, but in general :ok: 

> - created a dashboard_header, do we want to change them only in the public pages or in the dashboard , too? Inevitably the styles will change
> - left the links with a bit of opacity to keep the hover states, didn't found them in the mockups, is it white with underline, do I leave them like this?
> - do we lose "Learn" in favour of "Industries" + "Pricing" ?
> - I suppose the link "Login" was left with no update, applied the same style please confirm you want it different

I feel these questions are better suited for @CartoDB/design, it depends more on you ppl.

> - had to modify .UserAvatar-img in order to the signup button to be aligned. Avatar is 42px with that transparent border (what is it for?), button is 44px :/

Seems like the file history got b0rked with the common/old_common submodule move so can't verify, but I have some vague memory that it was due to some modifier rules had `border` set too, so this was added to maintain the same size. 
Anyhow, I feel we have a general problem with this kind of issue. Is there some argument for not applying a consistent `box-sizing`, like suggested by [in his article on css-tricks](https://css-tricks.com/box-sizing/), to avoid issues with borders/padidng affecting final element size? cc @xavijam ?

> - finally, I created .Header-navigationItem as we have .Header-settingsItem, actually, don't you think we should consolidate them both in a single class? the same with .Header-navigationLink

I'm all for consolidating, we have not been very good at this but we should do it more I think.

File: app/views/admin/shared/_dashboard_header.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <div class="Header-settings">
+      <ul class="Header-settingsList">
+        <% unless cartodb_com_hosted? %>
+        <li class="Header-settingsItem u-hideOnMobile">
+          <a href="http://cartodb.com/gallery/" class="Header-settingsLink">Explore</a>
+        </li>
+        <li class="Header-settingsItem js-login">
+          <a href="<%= CartoDB.url(self, 'login') %>" class="Header-settingsLink">Login</a>
+        </li>
+        <li class="Header-settingsItem js-learn" style="display:none;">
```
Comment (javierarce) [inline app/views/admin/shared/_dashboard_header.erb]: Does it make sense to have this style inline?

**Comment by javierarce** [comment]
:ok_hand: for me

File: app/views/admin/shared/_dashboard_header.erb (hunk 2 above)
Comment (matallo) [inline app/views/admin/shared/_dashboard_header.erb]: I'm :-1: for inlining but `display: none` to the component and then create a state `.is-active` but this is more work, for sure

(will do in following cases, will leave it like this as this is not the purpose of the issue, if you guys don't mind)

File: app/assets/stylesheets/variables/sizes.css.scss (hunk 1 above)
Comment (matallo) [inline app/assets/stylesheets/variables/sizes.css.scss]: because it's not a section, ha

but I see you point, so I'm changing it

in any case if we are reusing variables which semantically doesn't fit just because of the value, shouldn't we change its name?

Comment (viddo) [inline app/assets/stylesheets/variables/sizes.css.scss]: I added these definitions based on the styleguide sketch file:
screen shot 2015-07-08 at 11 40 06
…with the intention to be consistent with how margins/paddings are set. 
But if we're not following these guidelines or values do not match these, perhaps it's just better to hardcode the values on the specific rules?

**Comment by xavijam** [comment]
About the box-sizing, if we want to set that for everything, perfect, but first we have the problem of the old CSS and we need to get rid of it first IMHO. This is one of the problems with "new" and "old" code.

**Comment by matallo** [comment]
ooh, I see, not changing that border, then

**Comment by carlostallon** [comment]
left the links with a bit of opacity to keep the hover states, didn't found them in the mockups, is it white with underline, do I leave them like this?
- the links in #FFF and :hover with .7 opacity

do we lose "Learn" in favour of "Industries" + "Pricing" ?
- We should have the same links than comercial site

I suppose the link "Login" was left with no update, applied the same style please confirm you want it different
- link color #FFF
- border color #FFF with .3 opacity and 2px width

had to modify .UserAvatar-img in order to the signup button to be aligned. Avatar is 42px with that transparent border (what is it for?), button is 44px :/
I unified all those sizes in a new mockup, 
- Avatar 36px
- Button 36px

Take a look to my proposal: 
captura de pantalla 2015-07-08 11 46 03

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5001/))

**Comment by matallo** [comment]
- made most of the work with the header, going on with the rest of the page
- I left ready in shared/dashboard_header the new dashboard header, but as I think it's not ready yet from the proposals (for example Org. cases, see attached) I'll focus on public pages only by now

screen shot 2015-07-15 at 17 44 55

- There's the notifications icon in the proposal but despite I managed to add the view I'll borrow some @vikingo time to ask him how to do this properly (we'll have to populate user model and LocalStorage)

**Comment by carlostallon** [comment]
Thanks Matallo!
I think this new nav is much more convenient, without no reference to the org, in the actual nav it's like a reminder with no functionality. I prefer to make the navigation to datasets and maps more accesible.

**Comment by matallo** [comment]
cool, then I'd include a reference to the org in the public page dropdown, as it's in the dashboard dropdown

what do you think @CartoDB/design?

dashboard
screen shot 2015-07-16 at 10 50 46

public page
screen shot 2015-07-16 at 10 51 00

**Comment by saleiva** [comment]
Are the links on the right the good ones?

**Comment by matallo** [comment]
maps and datasets? confirmed with @carlostallon

**Comment by carlostallon** [comment]
@matallo yep, I think is enough ; )

**Comment by saleiva** [comment]
I meant on the left,sorry

**Comment by matallo** [comment]
bump @carlostallon

**Comment by matallo** [comment]
I created a branch just for the header that could be reviewed and merged now

--

some of the things I've encountered (and as I know @viddo had some concerns) while I'm doing this

- better keep short class names
`.NotSupportedDialog-itemLink` => `.NotSupportedDialog-link`

- remove specificity when possible
`.PublicMap-map .Spinner` => `.Spinner--center`

- Aim for generic components
when creating the breadcrumb at first I went with `.PublicMap-breadcrumb` then made it `.Breadcrumb` then realized a similar component that wasn't really a breadcrumb, so again moved to `.Navmenu`

so summing up, I could have ended with something like `.PublicMap-breadcrumbItemLink` but settled on `.Navmenu-link` which would be reusable (thinking of dashboard) and is less verbose

I think I already asked this, but why is better rgba(#aaa, 1) than #aaa?

we are writing empty stylesheets for non-rendering CSS, worth investigating/improving?
`write public/assets/3.15.09/stylesheets/variables/colors.css (0.005s)`

**Comment by viddo** [comment]
> I think I already asked this, but why is better rgba(#aaa, 1) than #aaa?

AFAIK we might as well use #aaa in those cases, I think it only makes sense to use that when the opacity value is something else than 1
